### ISSUE TITLE
License missing

### ISSUE THREAD

**Issue by jkooij** [osrf/ros2multirobotbook#109]
Excellent work on putting together this book!

For a ROS2 university course, we are considering to adapt parts from this book (the ROS2 part only) and extending them with some exercises, e.g., from the ROS2 wiki, but also to our teaching environment and simulator. However, I do not see a License in the repository, nor in the generated book's content.

How is this repository licensed, specifically towards adaptation for education purposes (with attribution)?
Thanks!

**Comment by Yadunund** [comment]
Hi @jkooij,

It's really exciting to hear about your plans to incorporate some the contents here in your course!

Looks like this repo is missing a LICENSE declaration but i'd assume we would be okay with assigning it as Apache License 2.0. @mxgrey @mdfesto @gbiggs what do you think?

**Comment by jkooij** [comment]
Ah, great to see an initial positive response!

> It's really exciting to hear about your plans to incorporate some the contents here in your course!

Yes, in the past our course focused on ROS 1, for which we used [Jason M. O'Kane's free book "A Gentle Introduction to ROS"](https://jokane.net/agitr/). Now we are looking to (finally) switch the course to ROS 2, but we don't have a drop-in replacement for that book to cover the basics in ROS 2. While I was looking for alternatives, I found this repo which already contains a lot of good explanations of the core concepts and tools for ROS 2. Hopefully we can use it as the basis for our new lab manual.

**One consideration on the licensing though:**

I just noticed the official ROS 2 wiki & documentation repo, https://github.com/ros2/ros2_documentation , uses the [CC-BY-4.0 license](https://github.com/ros2/ros2_documentation/blob/rolling/LICENSE).

Do you happen to know if CC-BY-4.0 and Apache 2.0 are compatible?
In other words, would it be possible to combine parts from this book and the official ROS 2?

I found some sources that explain that CC-BY-SA-4.0 and Apache 2.0 are incompatible, but it is not so clear for CC-BY-4.0 and Apache 2.0. This [guide by Uni Pittsburgh](https://pitt.libguides.com/openlicensing/apache2) indicates that there could be concerns, but doesn't provide more detail.

**Comment by mdfesto** [comment]
Hi @jkooij, fantastic that you're moving the course over to ROS 2! 

Thanks for your patience. We won't take took long but we are reviewing this with a few stakeholders at the moment and will update shortly. We aim to find a way to make it as user-friendly and accessible as possible.

**Comment by gavanderhoorn** [comment]
Friendly ping?

And re: which license: Apache 2.0 is not often used for documentation. I would also expect CC for this.

**Comment by mxgrey** [comment]
Sorry for the delay, we had to have some internal discussions to figure out the right process for adding a license to a repo that already has many contributors. The process is underway with #111 and should be wrapped up by the end of next week at the latest.

**Comment by mxgrey** [comment]
The CC-BY-4.0 license is added now! Please feel free to reuse in accordance with the license!

**Comment by jkooij** [comment]
That's great, thanks a lot! Your work and that of your contributors is much appreciated!
